File: rb/features.rb
# BleachPoems SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module BleachPoemsFeatures
  def self.make_feature(name)
    case name
    when "base"
      BleachPoemsBaseFeature.new
    when "test"
      BleachPoemsTestFeature.new
    else
      BleachPoemsBaseFeature.new
    end
  end
end

File: py/features.py
# BleachPoems SDK feature factory

from feature.base_feature import BleachPoemsBaseFeature
from feature.test_feature import BleachPoemsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: BleachPoemsBaseFeature(),
        "test": lambda: BleachPoemsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
